Feature: Jetpack Activity Logs

kean opened the pull request:
## Project

https://linear.app/a8c/project/apple-dataview-in-activity-logs-747664472a7e

## Changes

- https://github.com/wordpress-mobile/WordPress-iOS/pull/24592
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24596
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24598
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24599
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24600
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24601
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24602
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24603
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24610
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24611

## Removals

- https://github.com/wordpress-mobile/WordPress-iOS/pull/24604
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24605
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24606
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24607
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24608
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24609

## Screenshots

<img width="320" alt="06" src="https://github.com/user-attachments/assets/610974a7-cb1c-496c-8f30-39b8ffb55cbc" />
<img width="320" alt="05" src="https://github.com/user-attachments/assets/01d2eb58-3631-4084-a856-7749294f8894" />

<br>

<img width="320" alt="04" src="https://github.com/user-attachments/assets/e727999d-188a-4f8c-a5d8-fe0d6e3a1f4f" />
<img width="320" alt="03" src="https://github.com/user-attachments/assets/f8f00b87-2ecd-43a0-96f8-408d07715c82" />

<br>

<img width="320" alt="02" src="https://github.com/user-attachments/assets/8593a02d-5959-4453-9712-200bdf368784" />
<img width="320" alt="01" src="https://github.com/user-attachments/assets/626fff85-7b19-4a4a-9d66-8ccde964e582" />

kean committed:
Add PaginatedResponse

kean committed:
Refactor PaginatedResponse to DataViewPaginatedResponse

- Rename PaginatedResponse to DataViewPaginatedResponse for better naming consistency
- Move files from Pagination to DataView directory
- Update documentation to focus on UI usage with PaginatedForEach
- Add comprehensive unit tests for DataViewPaginatedResponse

kean committed:
Fix deleteItem to update total count

- Update deleteItem method to properly decrement total when removing items
- Update tests to verify total count is correctly maintained after deletion

kean committed:
Rename LoadMoreFooterView

kean committed:
Add DataViewPaginatedForEach component

kean committed:
Add DataViewPaginatedResponseProtocol

kean committed:
Refactor subscribers to use DataViewPaginatedResponse

Replace custom SubscribersPaginatedResponse with generic DataViewPaginatedResponse system. Add notification-based subscriber deletion updates and improve pagination handling with DataViewPaginatedForEach component.

kean committed:
Refactor subscribers service creation and move deleteSubscriber extension

- Rename getSubscribersService() to maketSubscribersService() to better reflect factory pattern
- Move deleteSubscriber extension from SubsriberDetailsViewModel to dedicated SubscribersServiceRemote+Extensions file
- Update all callers to use the renamed method

kean committed:
Fix typo in Subscriber

kean committed:
Fix formatting

kean committed:
Fix typos and address comments

kean committed:
Merge branch 'trunk' into task/activity-logs-swiftui

kean committed:
Implement new Jetpack Activity Logs screen

kean committed:
Extract DataViewPaginatedResponse (#24592)

kean committed:
Implement Jetpack Activity Logs screen (list) (#24596)

dangermattic commented:
<!--
  0 Errors
  2 Warnings: View files have been modified,...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

kean committed:
Fix SwiftLint warnings

kean committed:
Add Miniature app

kean committed:
Add XcodeTarget_App as a dependency

kean committed:
Configure xcconfig (just use Reader)

kean committed:
Add Miniature App (#24598)

kean committed:
Add initial ActivityLogDetailsView

kean committed:
Add ActivityLogDetailsView to Miniature target

- Implement modern SwiftUI version of Activity Detail screen
- Use card-based design system matching SubscriberDetailsView
- Add proper localization with NSLocalizedString and reverse-DNS keys
- Use WPStyleGuide for consistent icon and color handling
- Include activity stats visualization for backup events
- Add restore and download backup actions with confirmation dialogs
- Integrate with existing Activity model and RewindStatus
- Update ContentView with navigation links for testing
- Configure MiniatureApp with proper navigation and theming

kean committed:
Extract Activity icon methods from WPStyleGuide to Activity extension

- Create Activity+Icon.swift with gridiconType, icon, and statusColor properties
- Update all references to use new Activity extension methods
- Deprecate old WPStyleGuide methods with proper deprecation messages
- Remove stringToGridiconTypeMapping from WPStyleGuide
- Fix linter warnings for trailing whitespace and shorthand optional binding

kean committed:
Remove rewindStatus parameter from ActivityLogDetailsView

- Remove rewindStatus parameter from ActivityLogDetailsView initializer
- Remove rewindStatus checks in ActivityActionsCard
- Update preview providers to not pass rewindStatus
- Update ContentView to not create mock RewindStatus instances
- Remove warning card for multisite installations
- Simplify restore button enable/disable logic

kean committed:
Dev

kean committed:
Add formattedContent

kean committed:
Move the code to the main target

kean committed:
Extract reusable ActivityActorAvatarView

kean committed:
Remove unused code and integrate ActivityLogDetailsView

kean committed:
Fix SwiftLint warnings

kean committed:
Add ActivityLogsDetailsView (#24599)

kean committed:
Add initial restore/download backup code

kean committed:
Remove JetpackSiteRef usages

kean committed:
Add multisite handling

kean committed:
Remove the new screens and use the existing flows

kean committed:
Add analytics

kean committed:
Cleanup

kean committed:
Add missing analytics

kean committed:
Cleanup

kean committed:
Fix an issue with not all restorable acitivies shown in backups

kean committed:
Fix an issue with restore/download flow layout

kean committed:
Show restore checkpoint in section

kean committed:
Extract reusable CardView and InfoRow components from SubscribersDetailsVIew

kean committed:
Cleanup

kean committed:
Add Restore and Download Backup buttons to ActivityLogDetailsView (#24600)

kean committed:
Add rewindable

kean committed:
Move restore buttons higher

kean committed:
Add date filtering back

kean committed:
Use ActivityLogsView to show Backups (#24601)

kean committed:
Add Backup tracking

kean committed:
Fix clear background in restore flows

kean committed:
Rework how we do polling

kean committed:
Rework how we manage backup statuss

kean committed:
Revert hasBackup change

kean committed:
Update tests

kean committed:
Add Downloadable Backup Section to Backups list (#24602)

kean committed:
Update release notes

kean committed:
Remove unused code

kean committed:
Cleanup

kean committed:
Remove Application from list

kean committed:
Update Dashboard to use new ActivityLogsView (#24603)

kean committed:
Remove BackupListViewController and use ActivityLogsViewController instead

- Delete BackupListViewController.swift and its extension
- Update showBackup() to use ActivityLogsViewController with isBackupMode
- Remove BackupListViewController reference from ActivityDetailViewController
- Implement displayBackupWithSiteID using ActivityLogsViewController
- Add WPAnalytics tracking to ContentCoordinator backup display

kean committed:
Remove JetpackActivityLogViewController and use ActivityLogsViewController

- Delete JetpackActivityLogViewController.swift
- Update showActivity() methods to use ActivityLogsViewController directly
- Update ActivityDetailViewController to check for ActivityLogsViewController
- Simplify DashboardActivityLogCardCell to use ActivityLogsViewController

kean committed:
Remove dataViews feature flag

- Remove dataViews case from FeatureFlag enum
- Remove dataViews from enabled property switch statement
- Remove dataViews description

The feature flag was already removed from usage in previous commits
where we removed JetpackActivityLogViewController and BackupListViewController.

kean committed:
Create separate BackupsViewController and remove isBackupMode from ActivityLogsViewController

- Create new BackupsViewController and BackupsView that reuses ActivityLogsView
- Remove isBackupMode parameter from ActivityLogsViewController
- Update all usages to use BackupsViewController for backups
- Update ActivityDetailViewController to recognize BackupsViewController presenter
- Keep ActivityLogsView and ActivityLogsViewModel unchanged as implementation details

kean committed:
Activity Logs: Code Removal – Part 1 (#24604)

kean committed:
Remove ActivityListViewModelTests

kean committed:
Remove ActivityTypeSelectorViewController

kean committed:
Remove ActivityListViewModel

kean committed:
Remove BaseActivityListViewController

kean committed:
Fix compilation

kean committed:
Activity Logs: Code Removal – Part 2 (#24605)

kean committed:
Remove RewindStatusRow

kean committed:
Remove RewindStatusTableViewCell

kean committed:
Remove ActivityListRow

kean committed:
Remove ActivityTableViewCell

kean committed:
Activity Logs: Code Removal – Part 3 (#24606)

kean committed:
Remove CalendarViewController

kean committed:
Remove JTAppleCalendar dependency

kean committed:
Activity Logs: Code Removal – Part 4 (#24607)

kean committed:
Remove FilterBarView

kean committed:
Remove FilterChipButton

kean committed:
Remove ActivityDetailViewController

kean committed:
Activity Logs: Code Removal – Part 5 (#24608)

kean committed:
Remove ActivityStore usages

kean committed:
Remove ActivityStoreTests

kean committed:
Fix how isAwaitingCredentials works

kean committed:
Remove ActivityStore

kean committed:
Activity Logs: Code Removal – Part 6 (#24609)

kean committed:
Integrate ActivityContentRouter and FormattableActivity

kean committed:
Integrate ActivityContentRouter and FormattableActivity (#24610)

kean committed:
Update filter icon

kean committed:
Update filters icon

kean committed:
Update icons for filters in DataViews (#24611)

kean committed:
Remove WPStyleGuide+Activity and fix an issue with ActivityFormattableContentView layout

kean committed:
Fix layout of ActivityFormattableContentView

kean committed:
Add placeholders for empty fields

kean committed:
Remove build instructions from CLAUDE.md

kean committed:
Add more instructions for CLAUDE

kean committed:
Handle a scenario where logs are from existing user

kean committed:
Remove obsolete tests

kean committed:
Update UI tests

kean committed:
Fix release build

kean committed:
Use Duration

crazytonyli commented on Modules/Sources/WordPressUI/Views/CardView.swift:
```diff
@@ -0,0 +1,56 @@
+                        .foregroundStyle(.secondary)
+                }
+                content()
+            }
```
Is this `Group` necessary? Does moving the `.frame` call to `VStack` and deleting this `Group` look as expected?

kean replied:
Yeah, it works the same – updated.

crazytonyli commented on WordPress/Classes/ViewRelated/Activity/List/ActivityLogsView.swift:
```diff
@@ -0,0 +1,153 @@
+    private func makeRow(with item: ActivityLogRowViewModel) -> some View {
+        ActivityLogRowView(viewModel: item)
+            .onAppear { response.onRowAppeared(item) }
+            .background {
```
This is the first time I've seen this. Is the intention to hide the chevron added by the `NavigationLink`?

kean replied:
It won't be necessary when we can switch to iOS 17+ https://stackoverflow.com/questions/77542126/swiftui-how-to-remove-chevron-on-navigation-link-list-in-2023. For now, this seems like the only and very common way to hide the chevrons. I added a TODO: to address it.

crazytonyli commented on WordPress/Classes/ViewRelated/Activity/List/ActivityTypeSelectionView.swift:
```diff
@@ -0,0 +1,169 @@
+
+    init(viewModel: ActivityLogsViewModel) {
+        self.viewModel = viewModel
+        self._selectedTypes = State(initialValue: viewModel.parameters.activityTypes)
```
Does instantiating `State` like has the same issue you mentioned in https://github.com/wordpress-mobile/WordPress-iOS/pull/24072#discussion_r1955188214 about `StateObject`?

kean replied:
In practice, doesn't seem so as it's presented modally, but I move it to `onAppear` just to be safe.

crazytonyli replied:
Could the `selectedTypes` changes get reverted because the `ActivityTypeSelectionView` is re-rendered (like going back to this view), which calls `.onAppear { selectedTypes = viewModel.parameters.activityTypes }` again?

crazytonyli commented on WordPress/Classes/ViewRelated/Activity/List/ActivityTypeSelectionView.swift:
```diff
@@ -0,0 +1,169 @@
+        }
+        .onAppear {
+            Task { await fetchActivityGroups() }
+        }
```
Use `.task`?

kean replied:
Done

kean committed:
Merge branch 'trunk' into logs-merge-test

kean committed:
Simplify CardView

kean committed:
Add TODO for iOS 17

kean committed:
Move state change to onAppear
